simtel_pipeline.py

```diff
@@ -0,0 +1,292 @@
+    ) = utils.generate_timing_mask(
+        time_integration_options['window_start'],
+        time_integration_options['window_width'],
+        peak_position)
+
+    # Image cleaning configuration
+    picture_threshold = options.picture_threshold
+    boundary_threshold = options.boundary_threshold
+    shower_distance = 200 * u.mm
+
+    # Filtering on big showers
+    min_photon = 50
+
+    # Config for Hillas parameters analysis
+    n_showers = 100000000
+    reclean = True
+
+    # Define the event stream
+    data_stream = event_stream(
+        filelist=file_list,
+        camera_geometry=digicam_geometry,
+        camera=digicam)
+
+    # create data_stream
+    # data_stream = hsm.hessio_event_source(
+    #    data,
+    #    camera_geometry=digicam_geometry,
+    #    camera=digicam)
+
+    # Clean pixels
+    data_stream = filter.set_pixels_to_zero(
+        data_stream, unwanted_pixels=pixel_not_wanted)
+
+    # Computation of baseline
+    #
+    # Methods:
+    #
+    # 'data'
+    # Baseline is computed as a mean of 'n_bins0' first time samples,
+    # 'n_bins1' last time samples.
+    # A key assumption in this method is that the shower in simulated
+    # data is somewhere in the middle of 50 samples.
+    # Each pixel in each event has its own baseline
+    #
+    # 'simtel'
+    # Baseline is taken as simulated value event.mc.pedestal/50 that can
+    # be set up with the use of variable 'fadc_pedestal'
+    # in CTA-ULTRA6-SST-DC.cfg.
+
+    data_stream = simtel_baseline.fill_baseline_r0(
+        data_stream, n_bins0=options.baseline_beginning,
+        n_bins1=options.baseline_end, method='data')
+
+    # Run the r1 calibration (i.e baseline substraction)
+    data_stream = r1.calibrate_to_r1(data_stream, dark_baseline)
+
+    # Run the dl1 calibration (compute charge in photons + cleaning)
+    data_stream = dl1.calibrate_to_dl1(
+        data_stream,
+        time_integration_options,
+        additional_mask=additional_mask,
+        picture_threshold=picture_threshold,
+        boundary_threshold=boundary_threshold)
+
+    # Return only showers with total number of p.e. above min_photon
+    data_stream = filter.filter_shower(data_stream, min_photon=min_photon)
+
+    # Suffix for output filenames
+    suffix = (
+              options.primary +
+              '_ze' + str(options.zenit_angle).zfill(2) +
+              '_az' + str(options.azimut).zfill(3) +
+              '_p' + str(options.picture_threshold).zfill(2) +
+              '_b' + str(options.boundary_threshold).zfill(2)
+             )
+
+    # Save cleaned events - pixels and corresponding values
+    filename_pix = 'pixels.txt'
+    filename_eventsimage = 'events_image_' + suffix + '.txt'
+    data_stream = events_image.save_events(
+        data_stream, directory + filename_pix,
+        directory + filename_eventsimage)
+
+    # Save simulated shower paramters
+    filename_showerparam = 'pipedmc_param_' + suffix + '.txt'
```

Comment:
At some point we should create an output format for the analysis.

Reply:
I completely agree with You. The final format of those files should be discussed and probably something better than a lot of TXT output files should be proposed..

Reply:
No need to define something *final*, I think. This sounds like, we have to use txt until we know **for sure** what will be final. But we will never know anything for sure ... Still we can make a little improvement any time we want to make one .. event when it is not final.